python/solid_dmft/postprocessing/maxent_gf_imp.py

```diff
@@ -117,59 +118,77 @@ def _run_maxent(gf_imp_tau, maxent_error, n_points_maxent, n_points_alpha,
     omega_mesh = HyperbolicOmegaMesh(omega_min=omega_min, omega_max=omega_max,
                                      n_points=n_points_maxent)
 
-    if not mpi.is_master_node():
-        return None, omega_mesh
-
-    # Initializes and runs the maxent solver
-    # TODO: parallelization over blocks
-    results = [{} for _ in range(len(gf_imp_tau))]
-    for i, block_gf in enumerate(gf_imp_tau):
-        print('-'*50 + '\nSolving impurity {}/{}\n'.format(i+1, len(gf_imp_tau)) + '-'*50)
-        print('Found blocks {}'.format(list(block_gf.indices)))
-        for block, gf in block_gf:
-            solver = PoormanMaxEnt(use_complex=True)
-            solver.set_G_tau(gf)
-            solver.set_error(maxent_error)
-            solver.omega = omega_mesh
-            solver.alpha_mesh = LogAlphaMesh(alpha_min=1e-6, alpha_max=1e2,
-                                             n_points=n_points_alpha)
-            results[i][block] = solver.run()
-
-            n_orb = gf.target_shape[0]
-            opt_alpha = np.zeros((n_orb, n_orb, 2), dtype=int)
-            opt_alpha[:,:,:] = -1 # set results to -1 to distinguish them from 0
-            for i_orb in range(n_orb):
-                for j_orb in range(n_orb):
-                    for l_com in range(2):  # loop over complex numbers
-                        if results[i][block].analyzer_results[i_orb][j_orb][l_com] == {}:
-                            continue
-                        opt_alpha[i_orb, j_orb, l_com] = results[i][block].analyzer_results[i_orb][j_orb][l_com][analyzer]['alpha_index']
-
-            mpi.report(f'Optimal alphas , block {block}:')
-            mpi.report('--- Real part ---', opt_alpha[:, :, 0])
-            if np.any(opt_alpha[:, :, 1] != -1):
-                mpi.report('--- Imag part ---', opt_alpha[:, :, 1])
-            if np.any(opt_alpha[:,:,0] == -1):
-                mpi.report('(a -1 indicates that maxent did not run for this block due to symmetry)')
-
-    return results, omega_mesh
-
-
-def _unpack_maxent_results(results, omega_mesh):
-    """
-    Converts maxent result to impurity list of dict with mesh
-    and spectral function from each analyzer.
-    """
-
-    data_linefit = [{key: r.get_A_out('LineFitAnalyzer') for key, r in block_res.items()}
-                    for block_res in results]
-    data_chi2 = [{key: r.get_A_out('Chi2CurvatureAnalyzer') for key, r in block_res.items()}
-                 for block_res in results]
+    mpi.report(f'Continuing impurities with blocks: ')
+    imps_blocks = []
+    for i, block_gf in enumerate(gf_imp_list):
+        blocks = list(block_gf.indices)
+        mpi.report('- Imp {}: {}'.format(i, blocks))
+        for block in blocks:
+            imps_blocks.append((i, block))
+    mpi.report('-'*50)
+    imps_blocks_indices = np.arange(len(imps_blocks))
+
+    # Initialize collective results
+    data_linefit = [np.zeros(n_points_maxent)] * len(imps_blocks)
```

Comment:
That's what confused me with the block size. Does it still work when you change the initialization to `[0] * len(imps_blocks)`? Because you anyway overwrite it later, I think it would be more robust to use an immutable type here, especially with the shallow copying of the list entry.

Reply:
I now used
```
 data_linefit = [0] * len(imps_blocks)     
 data_chi2 =  [0] * len(imps_blocks)       
```
which also works just fine for the init and the mpi gathering of data. Is that what you had in mind? The test were running just fine with that.

Reply:
Yes exactly, looks perfect. Thanks a lot for implementing this!